### PR TITLE
feat(api): add project file formats api support

### PR DESCRIPTION
**Pull request by innomaxx**

### CODE REVIEW

File: src/main/java/com/crowdin/client/projectsgroups/model/PropertyFileFormatSettings.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.crowdin.client.projectsgroups.model;
+
+import com.crowdin.client.core.model.EscapeQuotesMode;
+import com.crowdin.client.core.model.EscapeSpecialCharsMode;
+import lombok.Data;
+
+@Data
+public class PropertyFileFormatSettings extends FileFormatSettings {
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/projectsgroups/model/PropertyFileFormatSettings.java:8]:
```suggestion
public class PropertiesFileFormatSettings extends FileFormatSettings {
```

File: src/main/java/com/crowdin/client/projectsgroups/ProjectsGroupsApi.java [hunk 2]
```diff
@@ -170,4 +176,120 @@ public ResponseObject<? extends Project> editProject(Long projectId, List<PatchR
         ProjectResponseObject projectResponseObject = this.httpClient.patch(this.url + "/projects/" + projectId, request, new HttpRequestConfig(), ProjectResponseObject.class);
         return ResponseObject.of(projectResponseObject.getData());
     }
+
+    //<editor-fold desc="File Format Settings">
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/projectsgroups/ProjectsGroupsApi.java:180]: what is this?

File: src/main/java/com/crowdin/client/projectsgroups/ProjectsGroupsApi.java [hunk 3]
```diff
@@ -170,4 +176,120 @@ public ResponseObject<? extends Project> editProject(Long projectId, List<PatchR
...
+     * <li><a href="https://developer.crowdin.com/enterprise/api/v2/#operation/api.projects.file-format-settings.patch" target="_blank"><b>Enterprise API Documentation</b></a></li>
+     * </ul>
+     */
+    public ResponseObject<FileFormatSettingsResource> editProjectFileFormatSettings(Long projectId, Long fileFormatSettingsId, List<PatchRequest> request) throws HttpException, HttpBadRequestException {
+        String url = formUrl_projectFileFormatSettingsId(projectId, fileFormatSettingsId);
+        FileFormatSettingsResponseObject response = this.httpClient.patch(url, request, new HttpRequestConfig(), FileFormatSettingsResponseObject.class);
+        return ResponseObject.of(response.getData());
+    }
+
+    //<editor-fold desc="Helper methods">
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/projectsgroups/ProjectsGroupsApi.java:280]: ?

File: src/main/java/com/crowdin/client/projectsgroups/ProjectsGroupsApi.java [hunk 4]
```diff
@@ -170,4 +176,120 @@ public ResponseObject<? extends Project> editProject(Long projectId, List<PatchR
...
+    private String formUrl_projectFileFormatSettingsId(Long projectId, Long fileFormatSettingsId) {
+        return this.url + "/projects/" + projectId + "/file-format-settings/" + fileFormatSettingsId;
+    }
+
+    private String formUrl_projectFileFormatSettingsId_customSegmentations(Long projectId, Long fileFormatSettingsId) {
+        return this.url + "/projects/" + projectId + "/file-format-settings/" + fileFormatSettingsId + "/custom-segmentations";
+    }
+    //</editor-fold>
+
+    //</editor-fold>
```
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/projectsgroups/ProjectsGroupsApi.java:294]: ?

File: src/test/java/com/crowdin/client/core/http/EnumSerializationTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+public class EnumSerializationTest {
+
+    private final JsonTransformer jsonTransformer;
+
+    public EnumSerializationTest() {
+        this.jsonTransformer = new JacksonJsonTransformer();
+    }
+
+    //<editor-fold desc="Numeric enums">
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/client/core/http/EnumSerializationTest.java:21]: please check all such places

File: src/main/java/com/crowdin/client/projectsgroups/model/PropertyFileFormatSettings.java
Comment (andrii-bodnar) [inline src/main/java/com/crowdin/client/projectsgroups/model/PropertyFileFormatSettings.java:1]: The entire file and all usages should be renamed as well

Comment (innomaxx) [reply src/main/java/com/crowdin/client/projectsgroups/model/PropertyFileFormatSettings.java:1]: fixed

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**